core/shared/src/test/scala/org/scalacheck/PropSpecification.scala

```diff
@@ -269,6 +269,30 @@ object PropSpecification extends Properties("Prop") {
     })
   }
 
+  property("shrinking process always take the first value returned by the Shrink[T]") = {
```

Comment:
This test reproduces the issue. 

With the Shrink provided, normally the shrinking process will go like `12345678` -> `1234` -> `12` -> `1`.

With my previous PR, it checks  something like`12345678` -> `1234` -> `5678`, gets to the end of the stream provided by the custom Shrink and then finally returns `12345678`.  So it's not really the test hanging, but it shows a change in the way the value is shrinked. 

Modifying the stream to be infinite will create a test that never ends. I think that even with finite stream you can see some symptoms as every test will run through the entire stream provided by Shrink instead of choosing the first value.